mne/viz/utils.py

```diff
@@ -1297,3 +1316,63 @@ def _compute_scalings(scalings, inst):
         scale_factor = np.max(np.abs(scale_factor))
         scalings[key] = scale_factor
     return scalings
+
+
+class SelectFromCollection(object):
+    """Select indices from a matplotlib collection using `LassoSelector`.
+
+    Selected indices are saved in the `ind` attribute. This tool highlights
+    selected points by fading them out (i.e., reducing their alpha values).
+    If your collection has alpha < 1, this tool will permanently alter them.
+
+    Note that this tool selects collection objects based on their *origins*
+    (i.e., `offsets`).
+
+    Parameters
+    ----------
+    ax : :class:`~matplotlib.axes.Axes`
+        Axes to interact with.
+
+    collection : :class:`matplotlib.collections.Collection` subclass
+        Collection you want to select from.
+
+    alpha_other : 0 <= float <= 1
+        To highlight a selection, this tool sets all selected points to an
+        alpha value of 1 and non-selected points to `alpha_other`.
+    """
+
+    def __init__(self, ax, collection, ch_names, alpha_other=0.3):
+        from matplotlib.widgets import LassoSelector
+        self.canvas = ax.figure.canvas
+        self.collection = collection
+        self.ch_names = ch_names
+        self.alpha_other = alpha_other
+
+        self.xys = collection.get_offsets()
+        self.Npts = len(self.xys)
+
+        # Ensure that we have separate colors for each object
+        self.fc = collection.get_facecolors()
+        if len(self.fc) == 0:
+            raise ValueError('Collection must have a facecolor')
+        elif len(self.fc) == 1:
+            self.fc = np.tile(self.fc, self.Npts).reshape(self.Npts, -1)
+
+        self.lasso = LassoSelector(ax, onselect=self.onselect)
+        self.selection = list()
+
+    def onselect(self, verts):
+        from matplotlib.path import Path
+        path = Path(verts)
+        inds = np.nonzero([path.contains_point(xy) for xy in self.xys])[0]
+        self.selection = self.ch_names[inds]
+        self.fc[:, -1] = self.alpha_other
+        self.fc[inds, -1] = 1
+        self.collection.set_facecolors(self.fc)
+        self.canvas.draw_idle()
+
+    def disconnect(self):
+        self.lasso.disconnect_events()
+        self.fc[:, -1] = 1
+        self.collection.set_facecolors(self.fc)
+        self.canvas.draw_idle()
```

Comment:
here and above no dosctrings, a hint at the intention of each method would be a minimum